infoblox_netmri/__init__.py

```diff
@@ -76,6 +80,37 @@ def __init__(self, host, username, password, api_version="auto",
             raise ValueError("Incorrect API version")
 
     def _make_request(self, url, method="get", data=None, extra_headers=None):
+        """Prepares the request, checks for authentication and retries in case of issues
+
+        Args:
+            url (str): URL of the request
+            method (str): Any of "get", "post", "delete"
+            data (any): Possible extra data to send with the request
+            extra_headers (dict): Possible extra headers to send along in the request
+        Returns:
+            dict
+        """
+        retry = True
+        attempts = 0
+
+        while retry and attempts < 1:
```

Comment:
You don't really need retry. If the request is successful, you return. If not, you either retry, or you raise. So, retry never gets set to false.